Tests/SourceKitTests/ExecuteCommandTests.swift

```diff
@@ -0,0 +1,94 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import SKCore
+import SKSupport
+import SKTestSupport
+import XCTest
+
+@testable import SourceKit
+
+final class ExecuteCommandTests: XCTestCase {
+
+  /// Connection and lifetime management for the service.
+  var connection: TestSourceKitServer! = nil
+
+  /// The primary interface to make requests to the SourceKitServer.
+  var sk: TestClient! = nil
+
+  /// The server's workspace data. Accessing this is unsafe if the server does so concurrently.
+  var workspace: Workspace! = nil
+
+  override func tearDown() {
+    workspace = nil
+    sk = nil
+    connection = nil
+  }
+
+  override func setUp() {
+    connection = TestSourceKitServer()
+    sk = connection.client
+    _ = try! sk.sendSync(InitializeRequest(
+      processId: nil,
+      rootPath: nil,
+      rootURL: nil,
+      initializationOptions: nil,
+      capabilities: ClientCapabilities(workspace: nil, textDocument: nil),
+      trace: .off,
+      workspaceFolders: nil))
+
+    workspace = connection.server!.workspace!
+  }
+
+  func testCommandIsRoutedToTheCorrectServer() {
+    let haveClangd = ToolchainRegistry.shared.toolchains.contains { $0.clangd != nil }
+    if !haveClangd {
+      XCTFail("Cannot find clangd in toolchain")
+    }
```

Comment:
I actually wasn't able to get this or anything in LocalClangTests to pass, I haven't got the right toolchains yet.

Reply:
Thanks

Reply:
Thanks